internal/mode/static/licensing/collector_test.go

```diff
@@ -0,0 +1,125 @@
+package licensing_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/helpers"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/licensing"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/dataplane"
+)
+
+var _ = Describe("DeploymentContextCollector", func() {
+	var (
+		clusterID = "test-uid"
+		ngfPod    = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod1",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind: "ReplicaSet",
+						Name: "replicaset1",
+					},
+				},
+			},
+		}
+
+		ngfReplicaSet = &appsv1.ReplicaSet{
+			Spec: appsv1.ReplicaSetSpec{
+				Replicas: helpers.GetPointer[int32](1),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "replicaset1",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind: "Deployment",
+						Name: "Deployment1",
+						UID:  "test-uid-replicaSet",
+					},
+				},
+			},
+		}
+
+		kubeNamespace = &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: metav1.NamespaceSystem,
+				UID:  "test-uid",
+			},
+		}
+
+		nodeList = &v1.NodeList{
+			Items: []v1.Node{{}},
+		}
+	)
+
+	It("collects the deployment context", func() {
+		collector := licensing.NewDeploymentContextCollector(licensing.DeploymentContextCollectorConfig{
+			K8sClientReader: fake.NewFakeClient(ngfPod, ngfReplicaSet, kubeNamespace, nodeList),
+			PodNSName:       types.NamespacedName{Name: ngfPod.Name},
+		})
+
+		expCtx := dataplane.DeploymentContext{
+			Integration:      "ngf",
+			ClusterID:        clusterID,
+			InstallationID:   "test-uid-replicaSet",
+			ClusterNodeCount: 1,
+		}
+
+		depCtx, err := collector.Collect(context.Background())
+		Expect(err).ToNot(HaveOccurred())
+		Expect(depCtx).To(Equal(expCtx))
+	})
+
+	It("returns an error if cluster info isn't found", func() {
+		collector := licensing.NewDeploymentContextCollector(licensing.DeploymentContextCollectorConfig{
+			K8sClientReader: fake.NewFakeClient(),
+		})
+
+		_, err := collector.Collect(context.Background())
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("error getting cluster information"))
+	})
+
+	It("returns the deployment context when the replicaset isn't found", func() {
```

Comment:
Can we test that the correct errors were logged? For this and the test case below?

Reply:
The logging has been removed from this function, but in general we don't test logs. Ideally, we shouldn't handle errors by logging. Instead, we should return the error to the caller and let the caller decide what to do with it.